In [1]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [2]:
!pip install python-dateutil
!pip install vaderSentiment
!pip install transformers
!pip install googletrans

In [3]:
import  os
import dateutil.parser as parser
import pandas as pd
import numpy as np
import re
import nltk
import pandas as pd
from textblob import TextBlob
from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer
from googletrans import Translator
from transformers import pipeline
import IPython


In [4]:
ls '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv'

Burgerology_comment_data.csv          manikfoodsdhanmondi_response_data.csv
Burgerology_post_data.csv             mrburgerbd_post_data.csv
Burgerology_response_data.csv         mrburger_comment_data.csv
Chillox_comment_data.csv              mrburger_response_data.csv
Chillox_post_data.csv                 Takeout_comment_data.csv
Chillox_response_data.csv             Takeout_post_data.csv
manikfoodsdhanmondi_comment_data.csv  Takeout_response_data.csv
manikfoodsdhanmondi_post_data.csv


In [5]:
base = '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/'

In [6]:
all_df  = [ base+'Burgerology_comment_data.csv',
           base+'Burgerology_response_data.csv',  
           
           base+'manikfoodsdhanmondi_comment_data.csv',
           base+'manikfoodsdhanmondi_response_data.csv',
                    
           base+'mrburger_comment_data.csv',
           base+'mrburger_response_data.csv',

           base+'Chillox_comment_data.csv',
           base+'Chillox_response_data.csv',             
             
           base+'Takeout_comment_data.csv',
           base+'Takeout_response_data.csv'
]

post_df  = [  base+'Burgerology_post_data.csv',        
             base+'mrburgerbd_post_data.csv',
             base+'Chillox_post_data.csv',           
             base+'Takeout_post_data.csv',
             base+'manikfoodsdhanmondi_post_data.csv'
]


comments_df  = [  base+'Burgerology_comment_data.csv',        
             base+'mrburger_comment_data.csv',
             base+'Chillox_comment_data.csv',           
             base+'Takeout_comment_data.csv',
             base+'manikfoodsdhanmondi_comment_data.csv'
]


response_df  = [ base+'Burgerology_response_data.csv',   
                 base+'mrburger_response_data.csv',
                base+'Chillox_response_data.csv', 
                base+'Takeout_response_data.csv',
             base+'manikfoodsdhanmondi_response_data.csv'        
]

In [7]:
all_df[0]

'/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Burgerology_comment_data.csv'

In [8]:
pd.read_csv(post_df[0]).head(2)

Unnamed: 0  ...                                             photos
0           0  ...  ['118971251_704734793443646_335789358787520516...
1           1  ...  ['118471658_698173437433115_162940010771481046...

[2 rows x 18 columns]

In [9]:
pd.read_csv(all_df[0]).head(2)

Unnamed: 0          post_id     type  ...  reactions img_name caption
0           0  704734973443628  comment  ...          1      NaN     NaN
1           1  704734973443628  comment  ...          1      NaN     NaN

[2 rows x 11 columns]

In [10]:
pd.read_csv(response_df[0]).head(2)

Unnamed: 0          post_id  ...  response_reaction reply_word_count
0           0  698176524099473  ...                  0                8
1           1  698176524099473  ...                  0                3

[2 rows x 11 columns]

# METRIC 1 → Brand Awareness

for demo purpose we will use Burgerology post's comment data

In [11]:
burgerology = pd.read_csv(comments_df[0])
burgerology = burgerology.text
tags_burgerology = ['burgerology','#burgerology']
tags_mrburger = ['mrburger','#mrburgerbd','mrburgerbd','mr. burger','mr.burger','mr. burgerbd','mr.burgerbd']
tags_chillox = ['chillox','#chillox']
tags_takeout = ['takeout','#takeout','bdtakeout','takeoutbd']
tags_manikfoodsdhanmondi = ['manikfoodsdhanmondi','#manikfoodsdhanmondi','manikfoods','#manikfoods','manik','#manik','mr.manik','mr manik']

# single pattern search

In [12]:
mention = []
count = 0
for i in range(len(burgerology)):
  ck = 0
  try:
    re.search('burgerology', burgerology[i].lower()).span()
    ck = 1
  except:
    pass

  if (ck):
    mention.append(burgerology[i])
    count+=1



In [13]:
count,len(burgerology)

(194, 2550)

well,from 2550 rows we have found 194 rows that contains the word "burgerology"

# Multiple pattern matching test

In [14]:
def brand_mentions(dfcomments,tags):
  mentions = []
  count = 0
  dfcomments = dfcomments.values.tolist()
  for i in range(len(dfcomments)):
    #ck = len(re.findall(r"\bburgerology\b|\b#burgerology\b", burgerology[i].lower()))
    #ck = len(re.findall(r"\b%s\b|\b%s\b"%(tags_burgerology[0],tags_burgerology[1]), burgerology[i].lower()))
    ck = 0
    for  j  in range(len(tags)):
      if (re.search(tags[j],  dfcomments[i].lower())):
        ck = 1

    if (ck):
      mentions.append(dfcomments[i])
      count+=ck
  return count

In [15]:

all_df

['/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Burgerology_comment_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Burgerology_response_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/manikfoodsdhanmondi_comment_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/manikfoodsdhanmondi_response_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/mrburger_comment_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/mrburger_response_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Chillox_comment_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Chillox_response_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Takeout_comment_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Takeout_response_data.csv']

In [16]:
df1 = pd.read_csv(all_df[0])
df2 = pd.read_csv(all_df[1])
Burgerology = pd.concat([df2.response_text, df1.text])

df1 = pd.read_csv(all_df[2])
df2 = pd.read_csv(all_df[3])
manikfoods = pd.concat([df2.response_text, df1.text])

df1 = pd.read_csv(all_df[4])
df2 = pd.read_csv(all_df[5])
mrburger = pd.concat([df2.response_text, df1.text])

df1 = pd.read_csv(all_df[6])
df2 = pd.read_csv(all_df[7])
Chillox = pd.concat([df2.response_text, df1.text])

df1 = pd.read_csv(all_df[8])
df2 = pd.read_csv(all_df[9])
Takeout = pd.concat([df2.response_text, df1.text])


In [17]:
burgerology_count = brand_mentions(Burgerology,tags_burgerology)
manikfoodsdhanmondi_count = brand_mentions(manikfoods,tags_manikfoodsdhanmondi)
takeout_count = brand_mentions(Takeout,tags_takeout)
chillox_count = brand_mentions(Chillox,tags_chillox)
mrburger_count = brand_mentions(mrburger,tags_mrburger)

# compare result with other brands(example : mrburgerbd)

In [18]:
print("Total Mentions of burgerology = ",burgerology_count,"from  total comments of = ", len(Burgerology))
print("Total Mentions of manikfoodsdhanmondi = ",manikfoodsdhanmondi_count,"from  total comments of = ", len(manikfoods))
print("Total Mentions of takeout = ",takeout_count,"from  total comments of = ", len(Takeout))
print("Total Mentions of chillox = ",chillox_count,"from  total comments of = ", len(Chillox))
print("Total Mentions of mrburger = ",mrburger_count,"from  total comments of = ", len(mrburger))

Total Mentions of burgerology =  212 from  total comments of =  2679
Total Mentions of manikfoodsdhanmondi =  56 from  total comments of =  3045
Total Mentions of takeout =  451 from  total comments of =  6746
Total Mentions of chillox =  226 from  total comments of =  5226
Total Mentions of mrburger =  45 from  total comments of =  1813


In [19]:
import plotly.graph_objects as go
colors = ['gold', 'mediumturquoise']
layout = {
  'title': 'Total Mentions of each Brand'
};
fig = go.Figure(data=[go.Pie(labels=['burgerology','manikfoodsdhanmondi','takeout','chillox','mrburger'],
                             values=[burgerology_count, manikfoodsdhanmondi_count,takeout_count,chillox_count,mrburger_count])],layout=layout)
fig.update_traces(hoverinfo='label+percent', textinfo='value', textfont_size=20,
                  marker=dict(colors=colors, line=dict(color='#000000', width=2)))
fig.show()

In [20]:
burgerolog = pd.read_csv(comments_df[0])


In [21]:
burgerologyposts = pd.read_csv(post_df[0])
mrburgerbdposts = pd.read_csv(post_df[1])
chilloxposts = pd.read_csv(post_df[2])
takeoutposts = pd.read_csv(post_df[3])
mrmanikposts = pd.read_csv(post_df[4])


#total shares



In [22]:
print("mrburgerbdposts = ",mrburgerbdposts.shares.sum())
print("burgerologyposts =",burgerologyposts.shares.sum())
print("chilloxposts =",chilloxposts.shares.sum())
print("takeoutposts =",takeoutposts.shares.sum())
print("mrmanikposts =",mrmanikposts.shares.sum())

mrburgerbdposts =  426
burgerologyposts = 859
chilloxposts = 4232
takeoutposts = 1886
mrmanikposts = 353


In [23]:

colors = ['Green', 'Blue']
layout = {
  'title': 'Total SHARES of each Brand'
};
fig = go.Figure(data=[go.Pie(labels=  ["Mr. BurgerBd","Burgerology","chillox","Takeout", "mr.manik"] ,
                             values=[mrburgerbdposts.shares.sum(),burgerologyposts.shares.sum(),chilloxposts.shares.sum(),takeoutposts.shares.sum(),mrmanikposts.shares.sum()])],layout=layout)
fig.update_traces(hoverinfo='label+percent', textinfo='value', textfont_size=20,
                  marker=dict(colors=colors, line=dict(color='#000000', width=2)))
fig.show()

In [24]:
mrburgerbd = pd.read_csv(comments_df[1])

In [25]:
mrburger_mentions = []
count = 0
for i in range(len(mrburgerbd)):
  ck = len(re.findall(r"\bmrburger\b|\b#mrburgerbd\b|\bmrburgerbd\b|\bmr. burger\b|\bmr.burger\b|\bmr. burgerbd\b|\bmr.burgerbd\b", mrburgerbd.text[i].lower()))

  if (ck>0):
    mrburger_mentions.append(mrburgerbd.text[i])
    count+=1


In [26]:
count,len(mrburgerbd)

(38, 1640)

well,from 1640 rows we have found 38 rows that mentions Mr. BurgerBd which is very lower compared to "burgerology" mentions

In [27]:
post_df

['/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Burgerology_post_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/mrburgerbd_post_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Chillox_post_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Takeout_post_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/manikfoodsdhanmondi_post_data.csv']

In [28]:
def post_delay(df):
  gap = []
  for i in range(len(df) - 1):
    datetime_str1 = df.time[i]
    datetime_str2 = df.time[i+1]
    date1 = parser.parse(datetime_str1)
    date2 = parser.parse(datetime_str2)
    td = date1 - date2
    gap.append(td.days)
  return gap

In [29]:
delay1 = post_delay(mrburgerbdposts)
print("Mr. BurgerBd AVG POST DELAY(IN DAYS) = ",np.mean(delay1))

Mr. BurgerBd AVG POST DELAY(IN DAYS) =  7.423076923076923


In [30]:
delay2 = post_delay(burgerologyposts)
print("Burgerology AVG POST DELAY(IN DAYS) = ",np.mean(delay2))

Burgerology AVG POST DELAY(IN DAYS) =  3.6737967914438503


In [31]:
delay3 = post_delay(chilloxposts)
print("chillox AVG POST DELAY(IN DAYS) = ",np.mean(delay3))

chillox AVG POST DELAY(IN DAYS) =  2.804232804232804


In [32]:
delay4 = post_delay(takeoutposts)
print("Takeout AVG POST DELAY(IN DAYS) = ",np.mean(delay4))

Takeout AVG POST DELAY(IN DAYS) =  2.8197674418604652


In [33]:
delay5 = post_delay(mrmanikposts)
print("mr.manik AVG POST DELAY(IN DAYS) = ",np.mean(delay5))

mr.manik AVG POST DELAY(IN DAYS) =  10.088235294117647


In [34]:

colors = ['Red', 'Blue']
layout = {
  'title': 'AVG POST DELAY Rate(IN DAYS) for each Brand'
};
fig = go.Figure(data=[go.Pie(labels= ["Mr. BurgerBd","Burgerology","chillox","Takeout", "mr.manik"] ,
                             values=[np.mean(delay1),np.mean(delay2),np.mean(delay3),np.mean(delay4),np.mean(delay5)])],layout=layout)
fig.update_traces(hoverinfo='label+percent', textinfo='value', textfont_size=15,
                  marker=dict(colors=colors, line=dict(color='#000000', width=2)))
fig.show()

# METRIC 2 → Audience Growth Rate :

In [35]:
def growth_calculator(df):
  column_names = ["days", "shares"]
  sharedf = pd.DataFrame(columns = column_names)
  days = []
  shares = []
  for i in range(len(df) - 1):
    datetime_str1 = df.time[i]
    datetime_str2 = df.time[i+1]
    date1 = parser.parse(datetime_str1)
    date2 = parser.parse(datetime_str2)
    td = (date1 - date2)
    td = td.days
    days.append(td)
    shares.append(df.shares[i] + df.shares[i+1])
  sharedf['days'] = days
  sharedf['shares'] = shares

  return sharedf

In [36]:
sharedf = growth_calculator(burgerologyposts)
sharedf1 = growth_calculator(mrburgerbdposts)
sharedf2 = growth_calculator(mrmanikposts)
sharedf3 = growth_calculator(takeoutposts)
sharedf4 = growth_calculator(chilloxposts)

In [37]:
#sharedf = sharedf.sort_values(by=['days'], inplace=True)
import plotly.express as px

fig = px.histogram(sharedf, x = "days",y = "shares", title = 'Gap Between TWO consecutive posts(in days) VS Total Number of share of those TWO posts (BURGEROLOGY)',color="shares")
fig.show()

fig = px.histogram(sharedf1, x = "days",y = "shares", title = 'Gap Between TWO consecutive posts(in days) VS Total Number of share of those TWO posts (MR. BURGER BD)',color="shares")
fig.show()

fig = px.histogram(sharedf2, x = "days",y = "shares", title = 'Gap Between TWO consecutive posts(in days) VS Total Number of share of those TWO posts (MR. MANIK)',color="shares")
fig.show()

fig = px.histogram(sharedf3, x = "days",y = "shares", title = 'Gap Between TWO consecutive posts(in days) VS Total Number of share of those TWO posts (TAKEOUT)',color="shares")
fig.show()

fig = px.histogram(sharedf4, x = "days",y = "shares", title = 'Gap Between TWO consecutive posts(in days) VS Total Number of share of those TWO posts (CHILLOX)',color="shares")
fig.show()

# Metric 3 -> Applause Rate

In [38]:
def Applause_Rate(df,followers):
  like = df.like.sum()
  love = df.love.sum()
  care = df.care.sum()
  wow = df.wow.sum()
  total = like + love + care + wow
  
  return (total/followers) * 100


In [39]:
Burgerology_followers = 72106 #Burgerology
Burgerology_Applause_Rate = Applause_Rate(burgerologyposts,Burgerology_followers) 
Burgerology_Applause_Rate

21.0759160125371

In [40]:
mrburgerbd_followers = 50405 #mrburgerbd
mrburgerbd_Applause_Rate = Applause_Rate(mrburgerbdposts,mrburgerbd_followers) 
mrburgerbd_Applause_Rate

20.567404027378235

In [41]:
mrmanik_followers = 124893 #Mr.Manik Foods Ltd
mrmanik_Applause_Rate = Applause_Rate(mrmanikposts,mrmanik_followers) 
mrmanik_Applause_Rate

11.598728511605936

In [42]:
chillox_followers = 163392 #chillox
chillox_Applause_Rate = Applause_Rate(chilloxposts,chillox_followers) 
chillox_Applause_Rate

9.989473168820995

In [43]:
takeout_followers = 241448  #takeout
takeout_Applause_Rate = Applause_Rate(takeoutposts,takeout_followers) 
takeout_Applause_Rate

10.709552367383452

In [44]:

colors = ['pine', 'shamrock']
layout = {
  'title': 'Applause Rate of each Brand'
};
fig = go.Figure(data=[go.Pie(labels= ["Mr. BurgerBd","Burgerology","chillox","Takeout", "mr.manik"] ,
                             values=[mrburgerbd_Applause_Rate,Burgerology_Applause_Rate,chillox_Applause_Rate,takeout_Applause_Rate,mrmanik_Applause_Rate])],layout=layout)
fig.update_traces(hoverinfo='label+percent', textinfo='value', textfont_size=15,
                  marker=dict(colors=colors, line=dict(color='#000000', width=2)))
fig.show()

# Metric 4 -> Average Engagement Rate

In [45]:
def Average_Engagement_Rate(df,followers):
  like = df.like.sum()
  love = df.love.sum()
  care = df.care.sum()
  wow = df.wow.sum()
  shares = df.shares.sum()
  comments = df.comments.sum()
  haha = df.wow.sum()
  sad = df.wow.sum()
  angry = df.wow.sum()
  total = like + love + care + wow + shares + comments + haha + sad + angry
  return (total/followers) * 100


In [46]:
Burgerology_average_engagement_rate = Average_Engagement_Rate(burgerologyposts,Burgerology_followers)
Burgerology_average_engagement_rate

33.42440296230549

In [47]:
mrburgerbd_average_engagement_rate = Average_Engagement_Rate(mrburgerbdposts,mrburgerbd_followers)
mrburgerbd_average_engagement_rate

29.67562741791489

In [48]:
takeout_average_engagement_rate = Average_Engagement_Rate(takeoutposts,takeout_followers)
takeout_average_engagement_rate

20.437526920910507

In [49]:
chillox_average_engagement_rate = Average_Engagement_Rate(chilloxposts,chillox_followers)
chillox_average_engagement_rate

24.159077555816687

In [50]:
mrmanik_average_engagement_rate = Average_Engagement_Rate(mrmanikposts,mrmanik_followers)
mrmanik_average_engagement_rate

17.38848454276861

In [51]:

colors = ['purple', 'pink']
layout = {
  'title': 'Average Engagement Rate of each Brand'
};
fig = go.Figure(data=[go.Pie(labels= ["Mr. BurgerBd","Burgerology","chillox","Takeout", "mr.manik"] ,
                             values=[mrburgerbd_average_engagement_rate,Burgerology_average_engagement_rate,chillox_average_engagement_rate,takeout_average_engagement_rate,mrmanik_average_engagement_rate])],layout=layout)
fig.update_traces(hoverinfo='label+percent', textinfo='value', textfont_size=15,
                  marker=dict(colors=colors, line=dict(color='#000000', width=2)))
fig.show()

# Metric 5 -> Amplification Rate

**should be available in new fb look**

# Metric 6 -> Response delay rate  analysis

even  though it was not mentioned  in 18 metrics but i thought  it can be important  so  giving it a try,you can exclude this section/metric  if you think  it's not important

In [52]:
response_df

['/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Burgerology_response_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/mrburger_response_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Chillox_response_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Takeout_response_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/manikfoodsdhanmondi_response_data.csv']

In [53]:
def response_delay(responsedf):
  return responsedf.response_delay_min.sum() / len(responsedf.response_delay_min)

In [54]:
Burgerology_response_data = pd.read_csv(response_df[0])
mrburger_response_data = pd.read_csv(response_df[1])
Chillox_response_data = pd.read_csv(response_df[2])
Takeout_response_data = pd.read_csv(response_df[3])
manikfoodsdhanmondi_response_data = pd.read_csv(response_df[4])

In [55]:
print("AVG response delay of Burgerology = ",response_delay(Burgerology_response_data) ," minutes")
print("AVG response delay of mrburger = ",response_delay(mrburger_response_data) ," minutes")
print("AVG response delay of Chillox = ",response_delay(Chillox_response_data) ," minutes")
print("AVG response delay of Takeout = ",response_delay(Takeout_response_data) ," minutes")
print("AVG response delay of manikfoodsdhanmondi = ",response_delay(manikfoodsdhanmondi_response_data) ," minutes")

AVG response delay of Burgerology =  2402.8217054263564  minutes
AVG response delay of mrburger =  1207.9017341040462  minutes
AVG response delay of Chillox =  4876.818181818182  minutes
AVG response delay of Takeout =  1687.702807357212  minutes
AVG response delay of manikfoodsdhanmondi =  2472.1666666666665  minutes


In [56]:

colors = ['navy', 'coral']
layout = {
  'title': 'AVG response delay Rate (in minutes) of each Brand'
};
fig = go.Figure(data=[go.Pie(labels= ["Mr. BurgerBd","Burgerology","chillox","Takeout", "mr.manik"] ,
                             values=[response_delay(mrburger_response_data),response_delay(Burgerology_response_data),response_delay(Chillox_response_data),response_delay(Takeout_response_data),response_delay(manikfoodsdhanmondi_response_data)])],layout=layout)
fig.update_traces(hoverinfo='label+percent', textinfo='value', textfont_size=15,
                  marker=dict(colors=colors, line=dict(color='#000000', width=2)))
fig.show()

# Metric 7 -> Customer  Satisfaction

In [57]:
analyser = SentimentIntensityAnalyzer()
def sentiment_scores(text):
    vadersenti = analyser.polarity_scores(text)
    return pd.Series([vadersenti['pos'], vadersenti['neg'], vadersenti['neu'], vadersenti['compound']])

def textblob(text):
  sentence = text.split('।')
  en_sent = translator.translate(text).text
  en_split = en_sent.split('.')


  for i, val in enumerate(sentence):
      token = nltk.word_tokenize(val)
      

  for i, val in enumerate(en_split):
      tb = TextBlob(val)
      sentiment = tb.sentiment.polarity
      return sentiment

nltk.download('punkt')

translator = Translator()

[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Package punkt is already up-to-date!


In [58]:
def check_sentiments(df):
  yes = []
  no = []
  for i in range(len(df)):
    text = df.text[i]
    pos=neg=neu=0
    lang = translator.detect(text)
    confidence = lang.confidence
    langdetect = lang.lang
    if (langdetect == 'en'):

      res = sentiment_scores(text)
      if(res[3] >= 0.1):
        pos+=1
      elif (res[3] <= -0.1):
        neg+=1
      else:
        neu+=1
      

      sentiment = textblob(text)
      if (sentiment > 0):
              pos+=1
      elif (sentiment < 0):
          neg+=1
      else:
          neu+=1

    
      classify = pipeline('sentiment-analysis')
      pred = classify(text)
      if (pred[0]['label'] == 'NEGATIVE'):
        neg+=1
      elif (pred[0]['label'] == 'POSITIVE'):
        pos+=1

          

    elif (langdetect == 'bn'):
      b2e = translator.translate(text, dest='en')
      #"""""""""""""""""""""""""""""""""""""""Translated"""""""""""""""""""""""""""""""""""""""""'
      classify = pipeline('sentiment-analysis')
      pred = classify(b2e.text)

      if (pred[0]['label'] == 'NEGATIVE'):
        neg+=1
      elif (pred[0]['label'] == 'POSITIVE'):
        pos+=1
      
      #"""""""""""""""""""""""""""""""""""""""Vader On Translated"""""""""""""""""""""""""""""""""""""""""'


      res = sentiment_scores(b2e.text)
      if (res[3] >= 0.1):
        pos+=1
      elif (res[3] <= -0.1):
        neg+=1
      else:
        neu+=1
    

      #"""""""""""""""""""""""""""""""""""""""BANGLA"""""""""""""""""""""""""""""""""""""""""'
      classify = pipeline('sentiment-analysis')
      pred = classify(text)
      if (pred[0]['label'] == 'NEGATIVE'):
        neg+=1
      elif (pred[0]['label'] == 'POSITIVE'):
        pos+=1

      
      #"""""""""""""""""""""""""""""""""""""""TextBlob  On Bangla"""""""""""""""""""""""""""""""""""""""""'
      sentiment = textblob(text)
      if (sentiment > 0):
        pos+=1
      elif (sentiment < 0):
        neg+=1
      else:
        neu+=1
    

    else:
      #"""""""""""""""""""""""""""""""""""""""TextBlob  On Broken"""""""""""""""""""""""""""""""""""""""""'
      sentiment = textblob(text)
      if (sentiment > 0):
        pos+=1
      elif (sentiment < 0):
        neg+=1
      else:
        neu+=1


      classify = pipeline('sentiment-analysis')
      pred = classify(text)
      if (pred[0]['label'] == 'NEGATIVE'):
        neg+=1
      elif (pred[0]['label'] == 'POSITIVE'):
        pos+=1
      
    if (pos+neu >= 2):
      yes.append(1)
    else:
      no.append(1)
  pos=len(yes)/len(df.text) * 100
  neg = 100.0 - pos
  return pos,neg

In [59]:
comments_df

['/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Burgerology_comment_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/mrburger_comment_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Chillox_comment_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/Takeout_comment_data.csv',
 '/content/drive/My Drive/MarkoPoloAI/Demo/mobassir/scrappedcsv/manikfoodsdhanmondi_comment_data.csv']

In [60]:
Burgerology_comment = pd.read_csv(comments_df[0]).head(20)
mrburger_comment = pd.read_csv(comments_df[1]).head(20)
Chillox_comment = pd.read_csv(comments_df[2]).head(20)
Takeout_comment = pd.read_csv(comments_df[3]).head(20)
manikfoodsdhanmondi_comment = pd.read_csv(comments_df[4]).head(20)


In [61]:
%%time
Burgerology_pos,Burgerology_neg = check_sentiments(Burgerology_comment)
mrburger_pos,mrburger_neg = check_sentiments(mrburger_comment)
Chillox_pos,Chillox_neg = check_sentiments(Chillox_comment)
Takeout_pos,Takeout_neg = check_sentiments(Takeout_comment)
manikfoodsdhanmondi_pos,manikfoodsdhanmondi_neg = check_sentiments(manikfoodsdhanmondi_comment)


CPU times: user 4min 19s, sys: 15.6 s, total: 4min 35s
Wall time: 17min 14s


In [62]:
print(" Burgerology_pos sentiment ratio = ",Burgerology_pos,"\n Burgerology_neg sentiment ratio = ",Burgerology_neg)
print(" mrburger_pos sentiment ratio  = ",mrburger_pos,"\n mrburger_neg sentiment ratio = ",mrburger_neg)
print(" Chillox_pos sentiment ratio = ",Chillox_pos,"\n Chillox_neg sentiment ratio= ",Chillox_neg)
print(" Takeout_pos sentiment ratio = ",Takeout_pos,"\n Takeout_neg sentiment ratio = ",Takeout_neg)
print(" manikfoodsdhanmondi_pos sentiment ratio  = ",manikfoodsdhanmondi_pos,"\n manikfoodsdhanmondi_neg sentiment ratio = ",manikfoodsdhanmondi_neg)

 Burgerology_pos sentiment ratio =  85.0 
 Burgerology_neg sentiment ratio =  15.0
 mrburger_pos sentiment ratio  =  65.0 
 mrburger_neg sentiment ratio =  35.0
 Chillox_pos sentiment ratio =  60.0 
 Chillox_neg sentiment ratio=  40.0
 Takeout_pos sentiment ratio =  100.0 
 Takeout_neg sentiment ratio =  0.0
 manikfoodsdhanmondi_pos sentiment ratio  =  85.0 
 manikfoodsdhanmondi_neg sentiment ratio =  15.0


In [63]:
colors = ['lime', 'brown']
layout = {
  'title': 'Customer Satisfaction (Positive Sentiment Rate) of each Brand'
};
fig = go.Figure(data=[go.Pie(labels= ["Burgerology","Mr. BurgerBd","chillox","Takeout", "mr.manik"] ,
                             values=[Burgerology_pos,mrburger_pos,Chillox_pos,Takeout_pos,manikfoodsdhanmondi_pos])],layout=layout)

fig.update_traces(hoverinfo='label+percent', textinfo='value', textfont_size=15,
                  marker=dict(colors=colors, line=dict(color='#000000', width=2)))
fig.show()

In [64]:
colors = ['grey', 'amber']
layout = {
  'title': 'Customer Satisfaction (Negative Sentiment Rate) of each Brand'
};
fig = go.Figure(data=[go.Pie(labels= ["Burgerology","Mr. BurgerBd","chillox","Takeout", "mr.manik"] ,
                             values=[Burgerology_neg,mrburger_neg,Chillox_neg,Takeout_neg,manikfoodsdhanmondi_neg])],layout=layout)

fig.update_traces(hoverinfo='label+percent', textinfo='value', textfont_size=15,
                  marker=dict(colors=colors, line=dict(color='#000000', width=2)))
fig.show()

**NOTE : of course  the  stats  above  shouldn't be  trusted  at all  because  i  used  20 comments from each  dataframe and still it  takes  a  lot of time,it  was only made  for demo purpose  or  for sanity check only,just  to demonstrate  how  this metric can be  calculated easily,if  i  use  full  dataframe  then it will  take hours  and  then can possibly  run out of memory!**

# Metric 8 -> Response Reaction Rate

even though it was not mentioned within the list of 18 metrics but i thought it can be important so giving it a try,you can exclude this section/metric if you think it's not important

In [65]:
def response_reaction_rate(responsedf):
  return responsedf.response_reaction.sum() / len(responsedf.response_reaction)

In [66]:
print("AVG response reactions of Burgerology = ",response_delay(Burgerology_response_data) ," reactions")
print("AVG response reactions of mrburger = ",response_delay(mrburger_response_data) ," reactions")
print("AVG response reactions of Chillox = ",response_delay(Chillox_response_data) ," reactions")
print("AVG response reactions of Takeout = ",response_delay(Takeout_response_data) ," reactions")
print("AVG response reactions of manikfoodsdhanmondi = ",response_delay(manikfoodsdhanmondi_response_data) ," reactions")

AVG response reactions of Burgerology =  2402.8217054263564  reactions
AVG response reactions of mrburger =  1207.9017341040462  reactions
AVG response reactions of Chillox =  4876.818181818182  reactions
AVG response reactions of Takeout =  1687.702807357212  reactions
AVG response reactions of manikfoodsdhanmondi =  2472.1666666666665  reactions


In [68]:
colors = ['purple', 'blue']
layout = {
  'title': 'AVG response reactions of each Brand'
};
fig = go.Figure(data=[go.Pie(labels= ["Burgerology","Mr. BurgerBd","chillox","Takeout", "mr.manik"] ,
                             values=[response_delay(Burgerology_response_data),response_delay(mrburger_response_data),response_delay(Chillox_response_data),response_delay(Takeout_response_data),response_delay(manikfoodsdhanmondi_response_data)])],layout=layout)

fig.update_traces(hoverinfo='label+percent', textinfo='value', textfont_size=15,
                  marker=dict(colors=colors, line=dict(color='#000000', width=2)))
fig.show()

#other  mentioned metrics are not possible to implement with our present data